File: engine/engine-model/src/main/scala/tools/packet/message/ServerNotice.scala
package tools.packet.message

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ServerNotice(private var _type: Int, private var _channel: Int, private var _message: String, private var _smegaEar: Boolean) extends PacketInput {
  def theType: Int = _type

  def channel: Int = _channel

  def message: String = _message

  def smegaEar: Boolean = _smegaEar

  def this(_type: Int, _message: String) = {
    this(_type, 0, _message, false)
  }

  override def opcode(): SendOpcode = SendOpcode.SERVER_MESSAGE
}

File: engine/engine-model/src/main/java/tools/packet/message/ServerNotice.java
package tools.packet.message;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ServerNotice(Integer theType, Integer channel, String message, Boolean smegaEar) implements PacketInput {
   public ServerNotice(Integer theType, String message) {
      this(theType, 0, message, false);
   }

   @Override
   public SendOpcode opcode() {
      return SendOpcode.SERVER_MESSAGE;
   }
}